lib/src/log_manager.dart

```diff
@@ -103,24 +107,40 @@ class LogManager {
             });
   }
 
-  void log(String channel, LogMessageCallback messageCallback,
-      {LogDataCallback data, ToJsonEncodable toJsonEncodable}) {
+  void log(String channel, dynamic message,
+      {dynamic data, ToJsonEncodable toJsonEncodable}) {
     assert(channel != null);
     if (!shouldLog(channel)) {
       return;
     }
 
-    assert(messageCallback != null);
-    final String message = messageCallback();
     assert(message != null);
+    if (message is LogMessageCallback) {
```

Comment:
This is functional, but I think generally makes the API harder to use. You might consider instead:

- having the main logging API (the wrapper classes and the underlying `log()` impl not take callbacks
- having a separate performance oriented entrypoint lib (`lib/logs_debug.dart` ?), which exposes a version that uses callbacks. This version probably wouldn't need the nice wrapper class (`class Log`).